src/hr/hr.service.ts

```diff
@@ -0,0 +1,19 @@
+import { Injectable } from '@nestjs/common';
+import { HrCreateRequest, HrCreatedResponse } from 'src/types';
+
+import { Hr } from './entities/hr.entity';
+
+@Injectable()
+export class HrService {
+  async createHr(hrData: HrCreateRequest): Promise<HrCreatedResponse> {
```

Comment:
Musisz sobie spojrzeć na tworzenie Studenta i zrobić to analogicznie do tamtego.
Czyli: na początku tworzymy Usera na podstawie emaila. oraz roli (tutaj HR). Następnie tworzymy nowego HR tak jak zrobiłeś niżęj
` const hr = new Hr();`
Dodajemy pozostałe informacje: fullName, company, maxReservedStudents
następnie `await hr.save()`
kolejno podłączamy relacje z Userem
`hr.user= user`
i ostatecznie możesz dać `return hr.save()`

Reply:
Wydaje mi się, że wystarczy 1 save, ponieważ relacja user już istnieje w encji hr.

Reply:
Możesz mieć racje :) moje przyzwyczajenie